### PR TITLE
Feature/86/setup firebase

### PR DESCRIPTION
**Pull request by OlkaA**
Issue #86 
Setup firebase.
Changed webpack.config.js file to access variable for the client side.
Created context for firebase to check if user is authenticated or not.
 
Updated Home page as an example, to check if authentication and context work. (need to be deleted/changed afterwards).

### CODE REVIEW

**Review by laila1222 — APPROVED** [review]
Good job! 
When I register, the account appears on Firebase. 👍 
Context works too with logging in/out! 👍

File: src/client/components/Pages/Home/HomePage.js [hunk 1]
```diff
@@ -1,8 +1,104 @@
...
+        {({ isAuthenticated }) => {
+          return (
+            <div>
+              <section>Final project!</section>
+              <br />
+              <br />
+              <p>Am I signed In: {isAuthenticated ? "yes" : "no"}</p>
+
+              <br />
+              <br />
```
Comment (zkwsk) [inline src/client/components/Pages/Home/HomePage.js:56]: You are essentially using markup for "styling" here. The separation of concerns would be better if you add spacing with styling instead. Not a major issue, though, so I would not consider this blocking the issue.

File: src/client/context/AuthContext.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+export class Provider extends React.Component {
+  state = {
+    isAuthenticated: false
+  };
+
+  componentDidMount() {
+    firebase.auth().onAuthStateChanged(user => {
+      {
+        user && console.log("User", user.email);
```
Comment (zkwsk) [inline src/client/context/AuthContext.js:15]:
```suggestion
        // eslint-disable-next-line no-console
        user && console.log("User", user.email);
```

Please add the above comment to bypass the eslint rule that disallow use of console.log. We should generally not be using console.log, but in this case we are waiting for the private routes component to be finished, so it is a viable way to test things in the meantime, but as soon as my PR with ESlint rules is merged we will have errors on console.log statements if you don't explicitly state that this line should be ignored. 

Do the same for other console.log statements in this PR.

**Review by zkwsk — DISMISSED** [review]

**Comment by bytewiz** [comment]
@OlkaA There is an error in the build, let's fix that :)

File: src/client/context/AuthContext.js (hunk 2 above)
Comment (OlkaA) [reply src/client/context/AuthContext.js:15]: done

File: src/client/components/Pages/Home/HomePage.js (hunk 1 above)
Comment (OlkaA) [reply src/client/components/Pages/Home/HomePage.js:56]: As this part of the code will be changed anyway, so I have decided not to spend too much time on styling.
The code in HomePage component is shown as an example of using firebase.

**Review by bytewiz — APPROVED** [review]
Looks good 👍

**Pull request merged**